File: kotlin/rna-transcription/src/main/kotlin/RnaTranscription.kt
fun toRNA(dna: Char): Char = 
    when(dna) {
        'C' -> 'G'
        'G' -> 'C'
        'T' -> 'A'
        'A' -> 'U'
        else -> 'Z'
    }

fun transcribeToRna(dna: String): String = 
    if (dna == "") ""
    else dna.map { toRNA(it) }.joinToString("")

File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let mapRna (nucleotide: char) = 
    match nucleotide with
    | 'G' -> 'C'
    | 'C' -> 'G'
    | 'T' -> 'A'
    | 'A' -> 'U'

let toRna (dna: string): string = String.map mapRna dna